runtime/Go/antlr/lexer.go

```diff
@@ -25,11 +27,12 @@ type Lexer interface {
 type BaseLexer struct {
 	*BaseRecognizer
 
-	Interpreter         *LexerATNSimulator
+	Interpreter         ILexerATNSimulator
 	TokenStartCharIndex int
 	TokenStartLine      int
 	TokenStartColumn    int
 	ActionType          int
+	Virt                Lexer // The most derived lexer implementation. Allows virtual method calls.
```

Comment:
This sucks, but it's the only way AFAIK to simulate a virtual method call in Go. The implementor must pass the most derived type and set it as a field in the composed type.